### DISCUSSION TITLE
How to enable Nextcloud synchronization when icloudpd running in a Truenas scale docker container.

### DISCUSSION THREAD

**Discussion by mrkongster** [icloud-photos-downloader/icloud_photos_downloader#591] [Q&A]
I am interested in enabling the icloudpd downloads readable by Nextcloud or sync with Nextcloud.  My goal is to have a full copy of all of my icloud photos and videos on my Truenas scale server and use either Nextcloud photos 2.0 or photoprism (or others...) so I am not rely on Apple Photos and iCloud.  

I can't find any documentation on the following - this is under Optional Environment Variables section of https://hub.docker.com/r/boredazfcuk/icloudpd:

**trigger_nextlcoudcli_synchronisation**: This creates a file in the download directory after a new files are downloaded. My NextcloudCLI container will detect this and force an immediate sync to the Nextcloud server.

Can someone help?  I looked for this for about a week now and can't seem to find anything.  Thanks in advance.

**Comment by boredazfcuk** [comment]
Hi,

I deleted a bunch of old projects of mine from GitHub and DockerHub a while back. I'm not sure why that container (NextcloudCLI) got the chop, as I still use it and even made a few modifications to it a couple of months ago. I've re-created it on Dockerhub so you should be able to pull it again now: https://hub.docker.com/repository/docker/boredazfcuk/nextcloudcli/general

I'm not sure if you're seen the documentation for it yet... it lives here: https://github.com/boredazfcuk/docker-nextcloudcli

Basically, setting the `trigger_nextlcoudcli_synchronisation` variable will tell boredazfcuk/icloudpd to write a file to the home directory after new files are downloaded. The NextcloudCLI container watches for that file to appear and when it does, it triggers a sync. It also syncs on an interval so that it can maintain a copy of other folders outside of the icloud stuff.

For example, my Nextcloud and NextcloudCLI containers run on a single machine. The NextcloudCLI container syncs Documents/Desktop/Pictures/Videos/iCloud directories from the Nextcloud container. However, the Nextcloud container stores its database on Disk 1 and the NextcloudCLI container downloads files to Disk 2. That way I maintain two copies of my data on the same server, so it's protected against drive failure.

Any issues, just let me know.

**Reply by mrkongster** [reply]
Thank you for responding to my question, appreciate it.

**Reply by mrkongster** [reply]
Sorry in advance for the dumb question.  I need help to get the Nextcloud and icloudpd connected.  I have tried this on and off for months and recently installed latest version of Truenas Scale and icloudpd and Nextcloud+Collabra.  I have the following error and not sure how to fix it.  I also don't understand the instructions related to the nextcloud_target_dir.  I tried pointing the directory outside of the Nextcloud data folder and inside the Nextcloud data folder, both didn't change the outcome of the connectivity check failed error.

Question 1:
- **nextcloud_target_dir**: This is the name of the root folder that you want to place files in. Please note, it is not possible to upload files to the root of a Nextcloud server at this time.  [what does root folder referring to?]

Question 2:
- what is "Nextcloud connectivity check failed: 000" mean?
2023-11-10 10:20:38.418106-08:002023-11-10 11:20:38 INFO     All photos have been downloaded
2023-11-10 10:20:39.027228-08:002023-11-10 11:20:39 INFO     New files downloaded: 2
2023-11-10 10:20:39.315099-08:002023-11-10 11:20:39 INFO     Upload files to Nextcloud
2023-11-10 10:20:39.316050-08:002023-11-10 11:20:39 INFO     Checking Nextcloud connectivity...
2023-11-10 10:20:39.345585-08:002023-11-10 11:20:39 ERROR    Nextcloud connectivity check failed: 000


Question 3:
- how do I reset icloudpd so that the Question 2 error won't come up again even when I removed all environment variables related to nextcloud sync from icloudpd setup?

I included the log file as reference, not sure if it will help... 

Thank  you in advance
[2023.11.10 1133am icloudpd20231029_icloudpd20231029-ix-chart-5876849b5b-9qgvq_ix-chart.log](https://github.com/icloud-photos-downloader/icloud_photos_downloader/files/13322394/2023.11.10.1133am.icloudpd20231029_icloudpd20231029-ix-chart-5876849b5b-9qgvq_ix-chart.log)
[2023.11.10 - icloudpd20231029_icloudpd20231029-ix-chart-5876849b5b-9qgvq_ix-chart.log](https://github.com/icloud-photos-downloader/icloud_photos_downloader/files/13322395/2023.11.10.-.icloudpd20231029_icloudpd20231029-ix-chart-5876849b5b-9qgvq_ix-chart.log)
,
Anthony

**Comment by mrkongster** [comment]
I am not familiar with the docker, linux system so my question may be
pretty basic as I just don't know enough.  I installed the nextcloudcli
using the custom_app from Truechart.  The system is showing Active but I am
getting the error (file attached) for local directory not found.

My settings are:
nextcloud_user = admin
nextcloud_password = 123456
nextcloud_url = https://192.168.0.18:9001/remote.php/dav/files/admin/

My nextcloud dataset is: /mnt/main/nextcloud_data
My icloudpd photos are stored at: /mnt/main/photos/icloudphotos

Error message I got from nextcloudcli is:
2023-02-25 15:33:04.240871+00:00Sat Feb 25 08:33:04 2023 ERROR: Local
directory not mounted. /home/user/Nextcloud/.mounted does not exist - retry
in 5 minutes

Can you point me to more instructions on how to set this up?  I am very
sorry if I am asking dumb questions here.

Best regards,
Anthony


On Sun, Feb 19, 2023 at 5:06 PM boredazfcuk ***@***.***>
wrote:

> Hi,
>
> I deleted a bunch of old projects of mine from GitHub and DockerHub a
> while back. I'm not sure why that container (NextcloudCLI) got the chop, as
> I still use it and even made a few modifications to it a couple of months
> ago. I've re-created it on Dockerhub so you should be able to pull it again
> now:
> https://hub.docker.com/repository/docker/boredazfcuk/nextcloudcli/general
>
> I'm not sure if you're seen the documentation for it yet... it lives here:
> https://github.com/boredazfcuk/docker-nextcloudcli
>
> Basically, setting the trigger_nextlcoudcli_synchronisation variable will
> tell boredazfcuk/icloudpd to write a file to the home directory after new
> files are downloaded. The NextcloudCLI container watches for that file to
> appear and when it does, it triggers a sync. It also syncs on an interval
> so that it can maintain a copy of other folders outside of the icloud stuff.
>
> For example, my Nextcloud and NextcloudCLI containers run on a single
> machine. The NextcloudCLI container syncs
> Documents/Desktop/Pictures/Videos/iCloud directories from the Nextcloud
> container. However, the Nextcloud container stores its database on Disk 1
> and the NextcloudCLI container downloads files to Disk 2. That way I
> maintain two copies of my data on the same server, so it's protected
> against drive failure.
>
> Any issues, just let me know.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/icloud-photos-downloader/icloud_photos_downloader/discussions/591#discussioncomment-5027907>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6AAWVPQW4VNAWFC46Z2VDLWYKYPTANCNFSM6AAAAAAVBFRWE4>
> .
> You are receiving this because you authored the thread.Message ID:
> <icloud-photos-downloader/icloud_photos_downloader/repo-discussions/591/comments/5027907
> @github.com>
>


-- 
*Favorite quotes:*
"Life is a journey, not a guided tour." -- Unknown
"Success breeds complacency. Complacency breeds failure. Only the paranoid
survive." by Andy Grove

**Comment by mrkongster** [comment]
One more item I forgot to mention, my icloudpd log say
Nextcloud synchronisation trigger: Disabled

What's the command or setting I need to do to in icloudpd setup to enable
Nextcloud synchronisation trigger to enable?



On Sat, Feb 25, 2023 at 8:41 AM Anthony Kong ***@***.***> wrote:

> I am not familiar with the docker, linux system so my question may be
> pretty basic as I just don't know enough.  I installed the nextcloudcli
> using the custom_app from Truechart.  The system is showing Active but I am
> getting the error (file attached) for local directory not found.
>
> My settings are:
> nextcloud_user = admin
> nextcloud_password = 123456
> nextcloud_url = https://192.168.0.18:9001/remote.php/dav/files/admin/
>
> My nextcloud dataset is: /mnt/main/nextcloud_data
> My icloudpd photos are stored at: /mnt/main/photos/icloudphotos
>
> Error message I got from nextcloudcli is:
> 2023-02-25 15:33:04.240871+00:00Sat Feb 25 08:33:04 2023 ERROR: Local
> directory not mounted. /home/user/Nextcloud/.mounted does not exist - retry
> in 5 minutes
>
> Can you point me to more instructions on how to set this up?  I am very
> sorry if I am asking dumb questions here.
>
> Best regards,
> Anthony
>
>
> On Sun, Feb 19, 2023 at 5:06 PM boredazfcuk ***@***.***>
> wrote:
>
>> Hi,
>>
>> I deleted a bunch of old projects of mine from GitHub and DockerHub a
>> while back. I'm not sure why that container (NextcloudCLI) got the chop, as
>> I still use it and even made a few modifications to it a couple of months
>> ago. I've re-created it on Dockerhub so you should be able to pull it again
>> now:
>> https://hub.docker.com/repository/docker/boredazfcuk/nextcloudcli/general
>>
>> I'm not sure if you're seen the documentation for it yet... it lives
>> here: https://github.com/boredazfcuk/docker-nextcloudcli
>>
>> Basically, setting the trigger_nextlcoudcli_synchronisation variable
>> will tell boredazfcuk/icloudpd to write a file to the home directory after
>> new files are downloaded. The NextcloudCLI container watches for that file
>> to appear and when it does, it triggers a sync. It also syncs on an
>> interval so that it can maintain a copy of other folders outside of the
>> icloud stuff.
>>
>> For example, my Nextcloud and NextcloudCLI containers run on a single
>> machine. The NextcloudCLI container syncs
>> Documents/Desktop/Pictures/Videos/iCloud directories from the Nextcloud
>> container. However, the Nextcloud container stores its database on Disk 1
>> and the NextcloudCLI container downloads files to Disk 2. That way I
>> maintain two copies of my data on the same server, so it's protected
>> against drive failure.
>>
>> Any issues, just let me know.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/icloud-photos-downloader/icloud_photos_downloader/discussions/591#discussioncomment-5027907>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A6AAWVPQW4VNAWFC46Z2VDLWYKYPTANCNFSM6AAAAAAVBFRWE4>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> <icloud-photos-downloader/icloud_photos_downloader/repo-discussions/591/comments/5027907
>> @github.com>
>>
>
>
> --
> *Favorite quotes:*
> "Life is a journey, not a guided tour." -- Unknown
> "Success breeds complacency. Complacency breeds failure. Only the paranoid
> survive." by Andy Grove
>


-- 
*Favorite quotes:*
"Life is a journey, not a guided tour." -- Unknown
"Success breeds complacency. Complacency breeds failure. Only the paranoid
survive." by Andy Grove

**Reply by mrkongster** [reply]
FYI, I got the icloudpd Nextcloud sync trigger to Enable.
2023-02-25T16:02:46.186242095Z 2023-02-25 09:02:46 INFO     Nextcloud synchronisation trigger: Enabled

Environment variable I used:
Name: trigger_nextlcoudcli_synchronisation
Value: Enable

Still need assistant to get the Nextcloudcli and icloudpd link so my photos shows up in nextcloud.

**Comment by AndreyNikiforov** [comment]
@mrkongster Do you know if Nextcloud plugin for TrueNAS can "sense" file changes and synchronize automatically? I cloudpd would periodically synchronize assets from iCloud to TrueNas volume and Nextcloud would consider these changes as new/change Files automatically. I haven't used TrueNas or Nextcloud, so my question is from a zoomed-out view.

TrueNAS seems to [support running docker containers](https://www.truenas.com/docs/scale/scaletutorials/apps/docker/) in SCALE edition. I saw notes that ppl are running linux VM on TrueNAS and then docker in them. I'd like to know what is your final setup.

Alternatively to running docker, you can run [executables](https://github.com/icloud-photos-downloader/icloud_photos_downloader/releases/tag/v1.11.0). We do have linux binary and I don't know how/if it runs on FreeBSD/TrueNAS. You can also get source from this git repo and build/run it. Github service that we use to build executables [does not support freebsd](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), so I don't know a way to provide freebsd executables as part of release.

**Reply by mrkongster** [reply]
@AndreyNikiforov  
I am running TrueNAS-SCALE-22.12.1.  I am not sure which version of Linux it is based on.  I am running apps inside docker (I think this is how people say it).  Basically Truenas Scale have built-in app store and then I connected to Truechart to get more apps.  The Nextcloud app is already installed but looks like it is only looking at the dataset that it is pointed to.  I am not sure if I can point to additional dataset.  My Nextcloud is connected to /mnt/main/nextcloud_data while my iCloudpd is using /mnt/main/photos/icloudphotos.  I think Nextcloud will not be able to tell if I have new files in the /icloudphotos folder since it is not associated with it.  Hence, this is where (I think) nextcoudcli comes in to connect Nextcloud and iCloudpd.  But my challenge is not able to connect them.

Hope this help you understand my situation better.

Anthony

**Comment by boredazfcuk** [comment]
Is this another that's been logged in the wrong repo?

I notice a log snippet relating to my .mounted file fail-safe.

Just configure the Nextcloud variables, as per the CONFIGURATION.MD file, and it will upload everything that is downloaded: https://github.com/boredazfcuk/docker-icloudpd/blob/master/CONFIGURATION.md#nextcloud-configuration-variables

There are also a command line parameter to reupload your entire library, so you don't have to delete the icloudpd downloads: https://github.com/boredazfcuk/docker-icloudpd/blob/master/CONFIGURATION.md#command-line-parameters
